### DISCUSSION TITLE
Installing Skywalking on Kubernetes not successful

### DISCUSSION THREAD

**Discussion by roynilanjan** [apache/skywalking#10225] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Helm Chart (apache/skywalking-kubernetes)

### What happened

I tried installing skywalking on k8s cluster in the skywalking namespace following this link https://github.com/apache/skywalking-kubernetes . Except the UI pod all other pods failed to start as shown below :
pod/skywalking-es-init-tdwlh          0/1     Init:Error   0                2d21h
pod/skywalking-es-init-zxlc2          0/1     Init:Error   0                2d21h
pod/skywalking-es-init-n6rgr          0/1     Init:Error   0                2d21h
pod/skywalking-es-init-zbk22          0/1     Init:Error   0                26h
pod/skywalking-es-init-zhlmr          0/1     Init:Error   0                26h
pod/skywalking-es-init-p9bvr          0/1     Init:Error   0                26h
pod/skywalking-ui-67bcf65678-drp85    1/1     Running      3 (40m ago)      2d22h
pod/skywalking-oap-58d449f489-cztsh   0/1     Init:0/1     38 (7m4s ago)    2d22h
pod/skywalking-oap-58d449f489-d5xcq   0/1     Init:0/1     38 (6m52s ago)   2d22h
For Elasticsearch only master-0 pod is in running state as shown below 
pod/elasticsearch-master-1            0/1     Pending      0                2d22h
pod/elasticsearch-master-2            0/1     Pending      0                2d22h
pod/elasticsearch-master-0            0/1     Running      3 (40m ago)      2d22h

For storage on my cluster it is as below :
 k get sc
NAME                   PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
local-path (default)   rancher.io/local-path   Delete          WaitForFirstConsumer   false                  70d
standard               kubernetes.io/aws-ebs   Retain          Immediate              true                   2d22h

After port forwarding as below the UI does not show up in the browser for the url http://localhost:8080. It shows a blank page however the tab header title shows Apache skywalking
 kubectl port-forward svc/skywalking-ui 8080:80 -n skywalking
Forwarding from 127.0.0.1:8080 -> 8080
Forwarding from [::1]:8080 -> 8080

### What you expected to happen

1) All the pods under skywalking namespace should be in running or in a completed state.
2)  UI should show up on the browser after port forwarding.

### How to reproduce

export SKYWALKING_RELEASE_VERSION=4.3.0  
export SKYWALKING_RELEASE_NAME=skywalking  
export SKYWALKING_RELEASE_NAMESPACE=skywalking
export REPO=skywalking
helm repo add ${REPO} https://apache.jfrog.io/artifactory/skywalking-helm
helm install "${SKYWALKING_RELEASE_NAME}" ${REPO}/skywalking -n "${SKYWALKING_RELEASE_NAMESPACE}" \
  --set oap.image.tag=9.2.0 \
  --set oap.storageType=elasticsearch \
  --set ui.image.tag=9.2.0  

### Anything else

This issue happens when I try to install it on any kubernetes cluster. Currently I am running on k3s cluster.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
You at least should read init pod logs. It started failing, there are logs and reasons there.

**Reply by roynilanjan** [reply]
Pods are in a failed state and as a reason it shows waiting for elasticsearch

**Reply by wu-sheng** [reply]
Of course, it has to wait for ElasticSearch, the data is in it. So what happens to ElasticSearch deployment?

If you want us to help, provide more details, especially raw logs. It is impossible for us to guess.

**Comment by kezhenxu94** [comment]
You have already known the init job crashed because it's waiting for ElasticSearch, why don't you even just take a look at the ES pod status? They are also in Pending status.... which means you might have insufficient resources(memory, CPU or node number) for ES to start up, anyway this can be easily diagnosed by just looking into the Pending Pod's log/event.
